### PR TITLE
Add `mc_sgx_dcap_types::QuoteHeader`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #193** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/193" target="_blank">Graphite</a> 
    * **PR #194** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/194" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/194?utm_source=stack-comment).

File: dcap/types/src/quote_header.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    /// The attestation key type (algorithm)
+    pub fn attestation_key_type(&self) -> Result<Algorithm, FfiError> {
+        // It seems that SGX is inconsistent on the size of the key type.
+        // Some places use 32 bit and others use 16 bit
+        let key_type_32 = self.0.att_key_type as u32;
+        key_type_32.try_into()
+    }
+
+    /// The attestation key data
+    pub fn attestation_key_data(&self) -> u32 {
```
Comment (jcape) [inline dcap/types/src/quote_header.rs:28]: Nit: m0ar rusty
```suggestion
    pub fn attestation_key_raw(&self) -> u32 {
```

**Review by jcape — APPROVED** [review]
LGTM, minor naming nit

File: dcap/types/src/quote_header.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply dcap/types/src/quote_header.rs]: renamed as suggested

**Review by awygle — APPROVED** [review]
LGTM

**Comment by nick-mobilecoin** [comment]
Since the Quote header type has changed in each quote version it was decided that exposing it outside of the quote3 type will not provide much benefit.

**Pull request closed without merging**